gsva/table_visualization.R

```diff
@@ -0,0 +1,110 @@
+library(reshape2)
+library(ggplot2)
+library(pheatmap)
+
+
+# Rename the columns of a melted table
+rename.melted <- function(input, feature=NA, sample=NA, value=NA, in.place=FALSE) {
+    # Set up the data for use in-or-out-of place
+    if (in.place) {
+        data <- input
+    } else {
+        data <- data.frame(input)
+    }
+    cols <- colnames(data)
+    # Set the feature column
+    if (!is.na(feature)) {
+        if (!(feature %in% names(data))) {
+            stop(paste0('Error in table_visualization::preprocess - no column ',
+                        feature, ' in data'))
+        }
+        cols[cols==feature] <- 'Feature'
+    }
+    # Set the sample column
+    if (!is.na(sample)) {
+        if (!(sample %in% names(data))) {
+            stop(paste0('Error in table_visualization::preprocess - no column ',
+                        sample, ' in data'))
+        }
+        cols[cols==feature] <- 'Sample'
+    }
+    # Set the values column
+    if (!is.na(value)) {
+        if (!(value %in% names(data))) {
+            stop(paste0('Error in table_visualization::preprocess - no column ',
+                        value, ' in data'))
+        }
+        cols[cols==feature] <- 'Value'
+    }
+    # Assign the new column names
+    colnames(data) <- cols
+    return(data)
+}
+
+# Perform all necessary proprocessing steps before visualization
+melt <- function(input, col.as.sample=TRUE) {
+    melted <- melt(input)
+    colnames(melted) <- c('Feature','Sample','Value')
+    if (!col.as.samples) {
+        actuallySample <- melted$Feature
+        melted$Feature <- melted$Sample
+        melted$Sample <- actuallySample
+    }
+    return(melted)
+}
+
+# Produce a box and whisker plot for the cohort
+make.box <- function(data, sample, feature.name=NA, sample.name=NA, value.name=NA) {
+    df = data.frame(data)
+    cols <- colnames(df)
+    # Establish feature column name
+    if (is.na(feature.name)) {
+        if(!('Feature' %in% colnames(data))){
+            stop(paste0('Error in table_visualization::make.box - no Feature column in data with columns ',colnames(data)))
+        }
+    }
+    # Establish value column name
+    if (is.na(value.name)) {
+        if(!('Value' %in% colnames(data))){
+            stop(paste0('Error in table_visualization::make.box - no Value column in data with columns ',colnames(data)))
+        }
+    }
+    # Establish sample column name
+    if (is.na(sample.name)) {
+        if(!('Sample' %in% colnames(data))){
+            stop(paste0('Error in table_visualization::make.box - no Sample column in data with columns ',colnames(data)))
+        }
+    }
+    colnames(df)[cols==sample.name] <- "Sample"
+    colnames(df)[cols==value.name] <- "Value"
+    colnames(df)[cols==feature.name] <- "Feature"
```

Comment:
ggplot will use this as the axis label by default and "Feature" is not very informative for someone viewing the plots. Can the axis be relabeled based on input from user or can "Feature" be mapped to a more informative label in the parameters?

Reply:
The ggplot we create uses `+labs(x=value.name, y=feature.name)` from the function arguments, and only if no arguments were provided and the literal column names were "Feature" and "Value" are those used for the plot labels